src/net/ossl.cc

```diff
@@ -848,6 +848,9 @@ class session : public enable_shared_from_this<session>, public session_impl {
                 f = pull_encrypted_and_send();
             }
             return f.then([this]{
+                if (!_options.wait_for_eof_on_shutdown) {
+                    return make_ready_future<>();
+                }
```

Comment:
Can you add a note about why this is "sidestepped".

Reply:
This matches original behavior in the GnuTLS implementation